File: src/main/java/org/nia/logic/commands/Commands.java
package org.nia.logic.commands;

import org.nia.bots.CWTavernBot;
import org.nia.model.User;
import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
import org.telegram.telegrambots.meta.api.objects.Message;
import org.telegram.telegrambots.meta.api.objects.replykeyboard.ReplyKeyboardRemove;
import org.telegram.telegrambots.meta.api.objects.replykeyboard.ReplyKeyboardMarkup;
import org.telegram.telegrambots.meta.api.objects.replykeyboard.buttons.KeyboardRow;

import java.util.List;

/**
 * @author Иван, 12.03.2017.
 */
public interface Commands {

    public String apply(Message message, User from);

    public boolean isApplicable(Message message, User from);

    public default SendMessage getMessage(Message message, String answer) {
        SendMessage sendMessage = new SendMessage();
        sendMessage.setChatId(String.valueOf(message.getChat().getId()));
        //sendMessage.enableMarkdown(true);
        sendMessage.enableHtml(true);
        sendMessage.setText(answer);


        if (message.isUserMessage()) {
            ReplyKeyboardMarkup replyKeyboardMarkup = new ReplyKeyboardMarkup();
            replyKeyboardMarkup.setSelective(true);
            replyKeyboardMarkup.setResizeKeyboard(true);
            replyKeyboardMarkup.setOneTimeKeyboard(true);
            replyKeyboardMarkup.setKeyboard(CWTavernBot.getKeyboard(User.getFromMessage(message.getFrom())));
            sendMessage.setReplyMarkup(replyKeyboardMarkup);
        } else {
            ReplyKeyboardRemove replyKeyboardHide = new ReplyKeyboardRemove();
            replyKeyboardHide.setSelective(true);
            replyKeyboardHide.setRemoveKeyboard(false);
            sendMessage.setReplyMarkup(replyKeyboardHide);

        }

        return sendMessage;
    }

    public default SendMessage getPersonalMessage(User user, String answer) {
        SendMessage sendMessage = new SendMessage();
        sendMessage.setChatId(String.valueOf(user.getUserID()));
        //sendMessage.enableMarkdown(true);
        sendMessage.enableHtml(true);
        sendMessage.setText(answer);


        ReplyKeyboardMarkup replyKeyboardMarkup = new ReplyKeyboardMarkup();
        replyKeyboardMarkup.setSelective(true);
        replyKeyboardMarkup.setResizeKeyboard(true);
        replyKeyboardMarkup.setOneTimeKeyboard(true);
        List<KeyboardRow> keyboard = CWTavernBot.getKeyboard(user);
        if (keyboard != null && !keyboard.isEmpty()) {
            replyKeyboardMarkup.setKeyboard(keyboard);
        }
        sendMessage.setReplyMarkup(replyKeyboardMarkup);

        return sendMessage;
    }
}

File: src/main/java/ru/nia/tavern/service/commands/Commands.kt
package ru.nia.tavern.service.commands

import org.telegram.telegrambots.api.methods.send.SendMessage
import org.telegram.telegrambots.api.objects.Message
import org.telegram.telegrambots.api.objects.replykeyboard.ReplyKeyboardMarkup
import org.telegram.telegrambots.api.objects.replykeyboard.ReplyKeyboardRemove
import ru.nia.tavern.model.User
import ru.nia.tavern.service.CWTavernBot

/**
 * @author Иван, 12.03.2017.
 */
interface Commands {

    fun apply(message: Message, from: User): String

    fun isApplicable(message: Message, from: User): Boolean

    open fun getMessage(message: Message, answer: String): SendMessage {
        val sendMessage = SendMessage()
        sendMessage.setChatId(message.chat.id!!)
        //sendMessage.enableMarkdown(true);
        sendMessage.enableHtml(true)
        sendMessage.text = answer


        if (message.isUserMessage) {
            val replyKeyboardMarkup = ReplyKeyboardMarkup()
            replyKeyboardMarkup.selective = true
            replyKeyboardMarkup.resizeKeyboard = true
            replyKeyboardMarkup.oneTimeKeyboard = true
            val keyboard = CWTavernBot.getKeyboard(User.getFromMessage(message.from))
            if (keyboard != null && !keyboard.isEmpty()) {
                replyKeyboardMarkup.keyboard = keyboard
            }
            sendMessage.replyMarkup = replyKeyboardMarkup
        } else {
            val replyKeyboardHide = ReplyKeyboardRemove()
            replyKeyboardHide.selective = true
            sendMessage.replyMarkup = replyKeyboardHide

        }

        return sendMessage
    }

    fun getPersonalMessage(user: User, answer: String): SendMessage {
        val sendMessage = SendMessage()
        sendMessage.setChatId(user.userID.toLong())
        //sendMessage.enableMarkdown(true);
        sendMessage.enableHtml(true)
        sendMessage.text = answer


        val replyKeyboardMarkup = ReplyKeyboardMarkup()
        replyKeyboardMarkup.selective = true
        replyKeyboardMarkup.resizeKeyboard = true
        replyKeyboardMarkup.oneTimeKeyboard = true
        val keyboard = CWTavernBot.getKeyboard(user)
        if (keyboard != null && !keyboard.isEmpty()) {
            replyKeyboardMarkup.keyboard = keyboard
        }
        sendMessage.replyMarkup = replyKeyboardMarkup

        return sendMessage
    }
}
